share/spack/docker/package-index/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# prepare the package index in form of JSON files
+FROM ubuntu:18.04 AS build-env
+MAINTAINER Spack Maintainers <maintainers@spack.io>
+
+ENV SPACK_ROOT=/opt/spack             \
+    DEBIAN_FRONTEND=noninteractive
+
+COPY bin     ${SPACK_ROOT}/bin
+COPY etc     ${SPACK_ROOT}/etc
+COPY lib     ${SPACK_ROOT}/lib
+COPY share   ${SPACK_ROOT}/share
+COPY var     ${SPACK_ROOT}/var
+
+RUN apt-get -yqq update   \
+ && apt-get -yqq install  \
+        bash jq python    \
+ && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /build
+# single, large index file
+RUN ${SPACK_ROOT}/bin/spack list --format version_json > packages.json
+# individual packages split into a tree of :firstLetter/:packageName.json
+COPY share/spack/packages/split.sh ./
```

Comment:
I'm trying this PR and Docker says this file is missing:
```console
Step 11/17 : RUN ${SPACK_ROOT}/bin/spack list --format version_json > packages.json
 ---> Using cache
 ---> ac397f05ce75
Step 12/17 : COPY share/spack/packages/split.sh ./
COPY failed: stat /var/lib/docker/tmp/docker-builder269930503/share/spack/packages/split.sh: no such file or directory

$ ls share/spack/packages/
Dockerfile
```
Looking at the PR it seems it has been moved :thinking:

Reply:
Yes, please use the `-f` and `.` parameters while sitting in the spack root. learned this as well today.

Reply:
waaaiit... I forgot to commit my last changes. thanks for catching!